app/controllers/api/databases.php

```diff
@@ -2743,16 +2743,27 @@ function updateAttribute(
                 if (empty($related)) {
                     continue;
                 }
-                if (!\is_array($related)) {
-                    $related = [$related];
+
+                if (\is_array($related) && \array_values($related) === $related) {
```

Comment:
If we're doing this here and after the for loop, maybe it would be good to store this into a descriptive variable?